### ISSUE TITLE
handle_request doesn't always call handle_error

### ISSUE THREAD

**Issue by drewp** [toastdriven/itty#7]
It's possible to get this error:

Traceback (most recent call last):
  File "/usr/lib/python2.6/wsgiref/handlers.py", line 93, in run
    self.result = application(self.environ, self.start_response)
  File "lib/python2.6/site-packages/itty-0.6.4-py2.6.egg/itty.py", line 227, in handle_request
    return handle_error(e, request)

because this code uses 'request' on 227 when it might not be bound yet.

```
221     try:
222         request = Request(environ, start_response)
223         
224         (re_url, url, callback), kwargs = find_matching_url(request)
225         response = callback(request, **kwargs)
226     except Exception, e:
227         return handle_error(e, request)
```

I would expect line 222 to be before the try-block, but I also don't know how I made this error. It started happening when I sent a "Content-type: application/json" header with my POST request.

**Comment by toastdriven** [comment]
It's possible for the creation of the `Request` to fail in ways that can't be handled very well (WSGI server dependent). I'll add a little bit of code to make sure that at least you can get past this.

**Comment by toastdriven** [comment]
Fixed in SHA: dd007fc3ac3c271b1845667f694c25d524fdb32c
